Mlem/API/Internal/SiteVersion.swift

```diff
@@ -0,0 +1,200 @@
+//
+//  APISiteVersionNumber.swift
+//  Mlem
+//
+//  Created by Sjmarf on 09/09/2023.
+//
+import Foundation
+
+enum SiteVersionSuffix: Equatable {
+    /// The only type other than beta "-beta.x" and releaseCandidate "-rc.x" that I've seen
+    /// is lemmy.blahaj.zone's "-kt.x". I'm unable to determine the meaning of this.
+    case beta(Int), releaseCandidate(Int), other(String, Int? = nil)
+}
+
+extension SiteVersionSuffix: CustomStringConvertible {
+    var description: String {
+        switch self {
+        case .beta(let int):
+            return "beta.\(int)"
+        case .releaseCandidate(let int):
+            return "rc.\(int)"
+        case .other(let string, let int):
+            if let int = int {
+                return "\(string).\(int)"
+            }
+            return string
+        }
+    }
+}
+
+extension SiteVersionSuffix: Comparable {
+    static func < (lhs: SiteVersionSuffix, rhs: SiteVersionSuffix) -> Bool {
```

Comment:
FWIW semver just does alphanumeric sorting here https://semver.org/#spec-item-9

Reply:
Yeah, perhaps. I wasn't sure about lemmy.blahaj.zone's `-kt` identifier though - I'm not sure what it is, so I'm assuming it's blahaj's custom modified version of Lemmy? If that's the case, it should be considered higher than the main version. In contrast, Betas and RCs are considered lower than the main version, since they are pre-releases

Reply:
Yeah this is one of the examples they have:
`Example: 1.0.0-alpha < 1.0.0-alpha.1 < 1.0.0-alpha.beta < 1.0.0-beta < 1.0.0-beta.2 < 1.0.0-beta.11 < 1.0.0-rc.1 < 1.0.0.`

Reply:
`1.0.0-kt.1` should be greater than `1.0.0` in our case - but using the server sorting system, it would be less than `1.0.0`.